### ISSUE TITLE
Odd combobox behaviour....

### ISSUE THREAD

**Issue by SamSandq** [frang75/nappgui_src#86]
I create a Combo like this:

>     Combo *filter = combo_create();
>     layout_combo(filtLayout, filter, 3, 0);
>     combo_add_elem(filter, "1+ stars", NULL);
>     combo_add_elem(filter, "2+ stars", NULL);
>     combo_add_elem(filter, "3+ stars", NULL);
>     combo_add_elem(filter, "4+ stars ", NULL);
>     combo_add_elem(filter, "5 stars", NULL);
>     combo_add_elem(filter, "(no filter)", NULL);
>     combo_add_elem(filter, "(unrated)", NULL);

That works perfectly, and shows:

![Screenshot 2024-01-11 at 21 36 42](https://github.com/frang75/nappgui_src/assets/50418806/52e12f91-7056-4e1c-8253-abe97846586f)

But what I really want is this:

>     combo_add_elem(filter, "★%", NULL);
>     combo_add_elem(filter, "★★%", NULL);
>     combo_add_elem(filter, "★★★%", NULL);
>     combo_add_elem(filter, "★★★★%", NULL);
>     combo_add_elem(filter, "★★★★★", NULL);
>     combo_add_elem(filter, "(no filter)", NULL);
>     combo_add_elem(filter, "(unrated)", NULL);

which does not show correctly, as some elements are discarded:

![Screenshot 2024-01-11 at 21 37 25](https://github.com/frang75/nappgui_src/assets/50418806/6d6e8332-515b-4476-b7ab-235fc0345561)

What could be wrong?

**Comment by frang75** [comment]
Hi @SamSandq! I add this new case to the ComboBox review.

Related to: https://github.com/frang75/nappgui_src/issues/81 https://github.com/frang75/nappgui_src/issues/68

**Comment by SamSandq** [comment]
FYI, the popup control works perfectly, and in my case is also better, since I do not wish to be able to edit the values.

However, I noticed that there is no way to access the text of the selected popup choice, only the index. Perhaps a

> char_t *txt = popup_text(myPopup);

or

> char_t *txt = popup_text(myPopup, index);

could be considered.

**Comment by frang75** [comment]
Thanks for update. Therefore, I leave this issue open. Full review of ComboBox is mandatory.

You can get the text of selected Item in `OnSelect` event as part of `EvButton` event parameters. But also is a good idea to add `const char_t *popup_get_text(const PopUp *popup, const uint32_t index)` to public interface.
![popup_text](https://github.com/frang75/nappgui_src/assets/42999199/7f75c8ed-f492-4f90-969c-4fa92bc5cdc3)

**Comment by SamSandq** [comment]
Yes, I am using that to get the selected text. But what I am after is to
get the text of any item using its index.

Cheers,
Sam
---------------------------------
Sam


Den mån 15 jan. 2024 kl 06:19 skrev Francisco García Collado <
***@***.***>:

**Comment by frang75** [comment]
Added `popup_get_text()` https://github.com/frang75/nappgui_src/commit/8177e951cdb5a0d9670e7a499491d209a8dfbc9c.

**Comment by frang75** [comment]
Refactor, improvements and bugfixing of ComboBox in this commit: https://github.com/frang75/nappgui_src/commit/850954a50d2dfde51cd846dbb306cd9c6345c2dc
